### PR TITLE
RAT-436: checkstyle fixes for plugin

### PR DESCRIPTION
**Pull request by Claudenw**
Fix for RAT-436
Fixed checkstyle issues
    

- Marked deprecated code
- Removed deprecated code from checkstyle consideration
- Fixed checkstyle issues.
- Fixed javadoc issue.

### CODE REVIEW

File: apache-rat-plugin/src/main/java/org/apache/rat/mp/HeaderMatcherSpecification.java [hunk 1]
```diff
@@ -22,7 +22,9 @@
 /**
  * Used to specify additional license matchers. A license matcher is basically a class, which implements
  * {@link org.apache.rat.analysis.IHeaderMatcher}.
+ * @deprecated use Configuration file to define licenses
  */
+@Deprecated
 public class HeaderMatcherSpecification
```
Comment (ottlinger) [inline apache-rat-plugin/src/main/java/org/apache/rat/mp/HeaderMatcherSpecification.java:28]: This class contains unused methods? Is it reflection as well @Claudenw ?

Comment (ottlinger) [reply apache-rat-plugin/src/main/java/org/apache/rat/mp/HeaderMatcherSpecification.java]: Other methods in these classes are marked as unused as well - I assume it's all reflection and means missing tests ....

**Review by ottlinger — APPROVED** [review]
I assume the unused stuff is not so relevant here as the code is deprecated.

File: apache-rat-plugin/src/main/java/org/apache/rat/mp/HeaderMatcherSpecification.java (hunk 1 above)
Comment (Claudenw) [reply apache-rat-plugin/src/main/java/org/apache/rat/mp/HeaderMatcherSpecification.java]: Yes, these were called by entries in the pom.xml file.

**Pull request merged**